### ISSUE TITLE
"Flycheck", "Oneline" and "JSON" output formats display any issues that have been ignored by "disable-for-" comments

### ISSUE THREAD

**Issue by rickclare** [rrrene/credo#503]
### Precheck

* I believe this is a bug, and is present in 0.9.0-rc1 and above, but not in 0.8.x.  I've searched the open and closed Github issues and haven't found anything similar.

### Environment

* Credo version (`mix credo -v`): **v0.9.0-rc3** (bug also present in v0.9.0-rc1 and v0.9.0-rc2)
* Erlang/Elixir version (`elixir -v`): **Elixir 1.6.1 (compiled with OTP 20)**
* Operating system: **macOS (also occurs in Ubuntu 16.04)**

### What was I trying to do?

When running `mix credo` with the different output formats ("Default", "Flycheck", "Oneline" and "JSON") the "Default" format hides any issues that have been ignored the the "disable comments" (e.g. `# credo:disable-for-next-line`).  The other formats still print the issues to stdout, which was not the case with Credo v0.8

Although the exit status is `0` when running with any of these output formats (which is correct), the issues that have been disabled are being confused with other non-disabled issues (i.e. lots of noise). Therefore, when running `mix credo --flycheck` in our CI (or Atom Editor - linter-elixir-credo plugin) it is difficult to distinguish the actual issues from the "disabled issues".

For example, here is a (contrived) code snippet which has a "disable comment":

```
defmodule MyApp.Accounts.User do
  use Ecto.Schema

  schema "users" do
    field :email, :string
    field :password, :string
    timestamps()
  end

  def changeset(user, attrs) do
    # credo:disable-for-next-line
    cast(user, attrs, [:email, :password])
    |> validate_required([:email, :password])
  end
end
```

### Expected outcome

```
$ mix credo
Checking 57 source files ...

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.6 seconds (0.03s to load, 0.6s running checks)
139 mods/funs, found no issues.

$ mix credo --format json
{
  "issues": []
}

$ mix credo --format flycheck
# no displayed issues

$ mix credo --format oneline
# no displayed issues
```

### Actual outcome

```
$ mix credo
Checking 57 source files ...

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.6 seconds (0.03s to load, 0.6s running checks)
139 mods/funs, found no issues.

$ mix credo --format json
{
  "issues": [
    {
      "trigger": "TODO",
      "priority": 2,
      "message": "Pipe chain should start with a raw value.",
      "line_no": 12,
      "filename": "lib/my_app/accounts/user.ex",
      "column_end": null,
      "column": null,
      "check": "Credo.Check.Refactor.PipeChainStart",
      "category": "refactor"
    }
  ]
}

$ mix credo --format flycheck
lib/my_app/accounts/user.ex:12: F: Pipe chain should start with a raw value.

$ mix credo --format oneline
[F] → lib/my_app/accounts/user.ex:12 Pipe chain should start with a raw value.
```

**Comment by rrrene** [comment]
@rickclare Ha! That was an embarassing bug. 😱  Thx for reporting.

**Comment by rickclare** [comment]
@rrrene I'm not sure if I'm doing some wrong in my configuration, but I've tried upgrading to _master_ branch of Credo, to test your fix for the "disable comments" issue.  

On the master branch, it appears that the "disable comments" is not having effect on the printed output  in _any_ format mode (e.g. even in "default" format mode)

From my mix.exs file:

```elixir
...
{:credo, github: "rrrene/credo", runtime: false, only: [:dev, :test]},
...
```

Using example from the Credo README

```elixir
defp do_stuff() do
  # credo:disable-for-next-line
  IO.inspect {:we_want_this_inspect_in_production!}
end
```

And then running `mix credo`

```
$ mix credo
Checking 57 source files ...

  Warnings - please take a look
┃
┃ [W] ↗ There should be no calls to IO.inspect/1.
┃       lib/my_app/repo.ex:5:5 #(MyApp.Repo.do_stuff)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 0.8 seconds (0.2s to load, 0.6s running checks)
8 mods/funs, found 1 warning.

$ mix credo --format json 
{
  "issues": [
    {
      "trigger": "IO.inspect",
      "priority": 12,
      "message": "There should be no calls to IO.inspect/1.",
      "line_no": 5,
      "filename": "lib/my_app/repo.ex",
      "column_end": 15,
      "column": 5,
      "check": "Credo.Check.Warning.IoInspect",
      "category": "warning"
    }
  ]
}

$ mix credo --format oneline
[W] ↗ lib/my_app/repo.ex:5:5 There should be no calls to IO.inspect/1.

$ mix credo --format flycheck
lib/my_app/repo.ex:5:5: W: There should be no calls to IO.inspect/1.
```
